source/aws-bootstrap-kit/lib/organization.ts

```diff
@@ -14,145 +14,104 @@ See the License for the specific language governing permissions and
-          enableSSMAWSServiceAccess.node.addDependency(enableMultiAccountsSetupAWSServiceAccess);
-
-          //adding an explicit dependency as CloudFormation won't infer that calling listRoots must be done only when Organization creation is finished as there is no implicit dependency between the
-          //2 custom resources
-          root.node.addDependency(org);
-
-          this.rootId = root.getResponseField("Roots.0.Id");
-    }
+  /**
+   * The Id of the Organization
+   */
+  readonly id: string;
+
+  /**
+   * The Id of the root Organizational Unit of the Organization
+   */
+  readonly rootId: string;
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    let orgProvider = new OrganizationProvider(this, "orgProvider");
+
+    let org = new CustomResource(this, "orgCustomResource", {
+      serviceToken: orgProvider.provider.serviceToken,
+      resourceType: "Custom::OrganizationCreation",
+    });
+
+    this.id = org.getAtt("OrganizationId").toString();
+
+    let root = new cr.AwsCustomResource(this, "RootCustomResource", {
+      onCreate: {
+        service: "Organizations",
+        action: "listRoots",
+        physicalResourceId: cr.PhysicalResourceId.fromResponse("Roots.0.Id"),
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+      },
+      onUpdate: {
+        service: "Organizations",
+        action: "listRoots",
+        physicalResourceId: cr.PhysicalResourceId.fromResponse("Roots.0.Id"),
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+      },
+      onDelete: {
+        service: "Organizations",
+        action: "listRoots",
+        physicalResourceId: cr.PhysicalResourceId.fromResponse("Roots.0.Id"),
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+      },
+      installLatestAwsSdk: false,
+      policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
+        resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+
+    // Enabling SSM AWS Service access to be able to register delegated adminstrator
+    const enableSSMAWSServiceAccess = this.enableAWSServiceAccess("ssm.amazonaws.com");
+    const enableMultiAccountsSetupAWSServiceAccess = this.enableAWSServiceAccess("config-multiaccountsetup.amazonaws.com");
+
+    enableMultiAccountsSetupAWSServiceAccess.node.addDependency(org);
+    enableSSMAWSServiceAccess.node.addDependency(enableMultiAccountsSetupAWSServiceAccess);
+
+    //adding an explicit dependency as CloudFormation won't infer that calling listRoots must be done only when Organization creation is finished
+    //as there is no implicit dependency between the 2 custom resources
+    root.node.addDependency(org);
+
+    this.rootId = root.getResponseField("Roots.0.Id");
+  }
 
   private enableAWSServiceAccess(principal: string) {
```

Comment:
I would make this public, so it can be used to enable other services too

Reply:
We will probably change this logic of delegated administrator soon. Also, the Organization itself is not really public (it is in cdk-org as a standalone construct, but not in bootstrap-kit).